src/Compilers/CSharp/Portable/Binder/Imports.cs

```diff
@@ -576,6 +576,18 @@ private void Validate()
                 alias.Alias.CheckConstraints(semanticDiagnostics);
             }
 
+            foreach (var @using in Usings)
+            {
+                // Check if `using static` directives meet constraints.
+                if (@using.NamespaceOrType.IsType)
```

Comment:
Is it worth explicitly checking if this is a static using? Today yes when `NamespaceOrType.IsType` is true then it's a `using static`. That's possibly something that is not going to be true in the future though. Think it might benefit from a more explicit check. Possibly add a property to `NamespaceOrTypeUsingDirective` called `IsStatic` that does this under the hood.

Reply:
> Is it worth explicitly checking if this is a static using? 

It feels like it shouldn't matter whether the using is static or not, the constraints should be satisfied anyways.

---
In reply to: [227993769](https://github.com/dotnet/roslyn/pull/30737#discussion_r227993769) [](ancestors = 227993769)

Reply:
Not certain, but I'm leaning toward no because it seems like any conceivable feature that involves referencing a type in a using directive should still have the compiler check constraints on that type.

Reply:
Sounds good.